### PR TITLE
Updated messaging docs to include automatic backend selection

### PR DESCRIPTION
**Pull request by orangejenny**

### CODE REVIEW

File: docs/messaging/sms_backends.rst [hunk 1]
```diff
@@ -195,3 +195,28 @@ same originating number.
...
+    ('27', <SQLMobileBackend: Global Backend 'GRAPEVINE-ZA'>),
+    ('999', <SQLMobileBackend: Global Backend 'MOBILE_BACKEND_TEST'>),
+    ('1', <SQLMobileBackend: Global Backend 'MOBILE_BACKEND_TWILIO'>),
+    ('258', <SQLMobileBackend: Global Backend 'MOBILE_BACKEND_MOZ'>),
+    ('266', <SQLMobileBackend: Global Backend 'GRAPEVINE-ZA'>),
+    ('265', <SQLMobileBackend: Global Backend 'MOBILE_BACKEND_TWILIO'>),
+    ('91', <SQLMobileBackend: Global Backend 'MOBILE_BACKEND_UNICEL'>),
+    ('268', <SQLMobileBackend: Global Backend 'GRAPEVINE-ZA'>),
+    ('256', <SQLMobileBackend: Global Backend 'MOBILE_BACKEND_YO'>),
+    ('*', <SQLMobileBackend: Global Backend 'MOBILE_BACKEND_MACH'>)
```
Comment (mkangia) [inline docs/messaging/sms_backends.rst:222]: Should this mapping be in confluence instead, so that it can be updated easily when needed?

**Review by mkangia — APPROVED** [review]

File: docs/messaging/sms_backends.rst (hunk 1 above)
Comment (orangejenny) [reply docs/messaging/sms_backends.rst:222]: I see what you're saying, but only devs can update these mappings, so I think having the docs in github isn't especially burdensome.

Comment (mkangia) [reply docs/messaging/sms_backends.rst:222]: ya but isn't it relevant for users/GS to see?

Comment (orangejenny) [reply docs/messaging/sms_backends.rst:222]: This came up from an internal request, so I'd rather keep it in internal docs unless support fields this question often enough that they want it moved. For internal users I think it's okay to be here, in readthedocs.

Comment (mkangia) [reply docs/messaging/sms_backends.rst:222]: okay, thanks @orangejenny

**Pull request merged**